src/content/docs/dropins/checkout/tutorials/add-payment-method.mdx

```diff
@@ -0,0 +1,140 @@
+---
+title: Add a payment method
+description: Learn how to integrate third-party payment providers with the Commerce boilerplate template.
+---
+
+import Aside from '@components/Aside.astro';
+import FileTree from '@components/FileTree.astro';
+import { Tabs, TabItem, Code, Steps } from '@astrojs/starlight/components';
+import Tasks from '@components/Tasks.astro';
+import Task from '@components/Task.astro';
+
+The [Commerce boilerplate template](https://github.com/hlxsites/aem-boilerplate-commerce) supports a variety of third-party payment providers, enabling you to integrate the payment gateway of your choice.
+
+## Steps-by-step
+
+This tutorial provides a step-by-step guide for integrating the Adyen payment provider with the Commerce boilerplate template using a custom NPM package developed by Adobe. You can use these steps as a reference to integrate other payment providers as well.
+
+<Tasks>
+
+<Task>
+### Prerequisites
+
+You must [install and configure](https://docs.adyen.com/plugins/adobe-commerce/set-up-the-plugin-in-adobe-commerce) the Adyen Payment Plugin on your Adobe Commerce backend before integrating it with the Commerce boilerplate template.
+
+<Aside type="note" title="Compatibility">
+The Adyen package is not compatible with [API Mesh](https://developer.adobe.com/graphql-mesh-gateway/).
+</Aside>
+
+</Task>
+
+<Task>
+
+### Add your client key
+
+After you install and configure the Adyen Payment Plugin on your Adobe Commerce backend, you need to add your Adyen client key to the Commerce boilerplate configuration.
+
+<Aside type="note">
+  You should use a development or staging environment for integration testing purposes before modifying your production configuration.
+</Aside>
+
+<Steps>
+1. Open your project's `configs-stage` or `configs-dev` Google Sheet or Microsoft Excel file.
+
+1. Add a new key-value pair for your Adyen API key.
+
+   | key               | value           |
+   |-------------------|-----------------|
+   | adyen-client-key  | YOUR_CLIENT_KEY |
+
+1. Save the file.
+1. Publish file changes with the [Sidekick](/get-started/#sidekick) extension.
+   
+</Steps>
+
+</Task>
+
+<Task>
+
+### Install the package
+
+<Steps>
+1. Open a terminal and navigate to your boilerplate-based project.
+1. Install the Adyen NPM package and add it to the list of project dependencies in your `package.json` file with the `--save` flag.
+   
+   ```bash
+   npm install @dropins/adyen-checkout-extension --save
+   ```
+</Steps>
+
+</Task>
+
+<Task>
+
+### Integrate with the boilerplate
+
+To integrate third-party payment providers, like Adyen, with Adobe Commerce, you can use the extensibility features provided by the checkout drop-in component. The checkout component allows you to customize the list of payment methods shown during the checkout process using slots.
+
+<Steps>
+1. Open the `blocks/commerce-checkout/commerce-checkout.js` file.
+
+1. Import the required files.
+   
+   ```js
+   ...
+   // Adyen drop-in modules
```

Comment:
@sirugh, I thought I might be able to show the full code example and use some highlighting and callouts to describe the additions to the default `commerce-checkout.js` block file, but the file in the develop branch in the boilerplate repo has way too many lines. Do you have any suggestions for either using a smaller version of the default block file or providing more specific guidance about where the code additions described in these steps should go?

Reply:
The stripe example is probably the simplest example of how integration happens. The main point is just the payment slot. The rest is implementation details.

Reply:
https://github.com/hlxsites/boilerplate-commerce-dropins/pull/89/files#diff-abb36271baf859ae194f7cd3ef71819fe3772b798408056539d919f59fe42e8b

I based this integration on stripe's documentation and what I knew of the payment slot. It was not completed, note the TODOs.

But you can reference it to see a few things, such as how I use Stripe's SDK to create the elements, then inject them into the checkout using the payment slot's "addPaymentMethodHandler" function.

Within that function block I do some things like attach an "onPlaceOrder" handler which sets the method on the cart, and then actually places the order.

Still you can see there is some things like how I cannot call "completeCheckout", at least not back when I wrote this PR.

I think if Olena wants this documented, she needs to get engineering to dedicate some resources to "stabilizing"/refining either stripe or Adyen, otherwise we're going to be documenting POCs.